app/assets/javascripts/welcome.js.erb

```diff
@@ -1,121 +1,152 @@
-  var map, infoWindow;
-  var markers = [];
-  var savedmarkers= [];
-  var firstmarker= true;
-  var topic= '<%=image_url('0.png')%>';
-  function initMap() {
+var map, infoWindow;
+var markers = [];
+var markerstodelete = [];
+var firstmarker= true;
+var deletemarker = false;
+var topic= '<%=image_url('0.png')%>';
 
-    infoWindow = new google.maps.InfoWindow;
-
-      // HTML5 geolocation.
-      if (navigator.geolocation) {
-        navigator.geolocation.getCurrentPosition(function(position) {
-        var pos = {
-          lat: position.coords.latitude,
-          lng: position.coords.longitude
-        };
-
-        infoWindow.setPosition(pos);
-        infoWindow.setContent('Location found.');
-        infoWindow.open(map);
-        map.setCenter(pos);
-      }, function() {
-        handleLocationError(true, infoWindow, map.getCenter());
-      });
-    } else {
-      // Browser doesn't support Geolocation
-      handleLocationError(false, infoWindow, map.getCenter());
-    }
-    map = new google.maps.Map(document.getElementById('map'), {
+function initMap() {
+  infoWindow = new google.maps.InfoWindow;
+  map = new google.maps.Map(document.getElementById('map'), {
       zoom: 4,
-      center: {lat: -34.397, lng: 150.644},
+  });
+  // HTML5 geolocation.
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function(position) {
+    var pos = {
+      lat: position.coords.latitude,
+      lng: position.coords.longitude
+    };
+    infoWindow.setPosition(pos);
+    infoWindow.setContent('Location found.');
+    infoWindow.open(map);
+    map.setCenter(pos);
+    map.setZoom(16);
+    }, function() {
+      handleLocationError(true, infoWindow, map.getCenter());
     });
-    load_user_targets();
-    // This event listener calls addMarker() when the map is clicked.
-    google.maps.event.addListener(map, 'click', function(event) {
-      $.getJSON('/targets/load_create_target', function(data) {
-        $(".rectangle").html(data.form);
-        $('#latitude').val(event.latLng.lat());
-        $('#longitude').val(event.latLng.lng());
-        $(".topic").click(function() {
+  } else {
+  // Browser doesn't support Geolocation
+      handleLocationError(false, infoWindow, map.getCenter());
+  }
+  load_user_targets();
+  // This event listener calls addMarker() when the map is clicked.
+  google.maps.event.addListener(map, 'click', function(event) {
+    $.getJSON('/targets/load_create_target', function(data) {
+      $(".rectangle").html(data.form);
```

Comment:
In Javascript, try to always use single quotation marks. In this case: `$('.rectangle').html(data.form);`